src/parser/GraphPatternOperation.cpp

```diff
@@ -75,11 +77,13 @@ void BasicGraphPattern::appendTriples(BasicGraphPattern other) {
 }
 
 // ____________________________________________________________________________
-cppcoro::generator<const Variable> Bind::containedVariables() const {
+std::vector<Variable> Bind::containedVariables() const {
+  std::vector<Variable> output;
   for (const auto* ptr : _expression.containedVariables()) {
-    co_yield *ptr;
+    output.push_back(*ptr);
   }
-  co_yield _target;
+  output.push_back(_target);
+  return output;
```

Comment:
If you implement it in the header (otherwise you have to fiddle with the return value), 
you can do the very simple

```
auto containedVariables() const {
  return ad_utility::OwningView{_expression.containedVariables()} | ql::views::transform(ad_utility::dereference);
}
```
This avoids copying the variables.
}

Reply:
In a future PR I will update our ranges library s.t. we don't need the explicit `OwningView` thing anymore.

Reply:
> If you implement it in the header (otherwise you have to fiddle with the return value), you can do the very simple
> 
> ```
> auto containedVariables() const {
>   return ad_utility::OwningView{_expression.containedVariables()} | ql::views::transform(ad_utility::dereference);
> }
> ```
> 
> This avoids copying the variables. }

As this misses the target variable, I tried instead
`    auto expressionVariables =
        ad_utility::OwningView{_expression.containedVariables()} |
        ql::views::transform(ad_utility::dereference);

    return ql::views::concat(expressionVariables, ql::views::single(_target));`
But concat is not available in C++20. Any other idea which does not use actions? Otherwise we could stick to the vector proposal...

Reply:
Hi @dsciarra 
Thank you for your analysis, you are correct. 
We could have tried something like always using the `range-v3` version of `concat`, but that would have also lead to issues with the owning view for now (I have to apply some improvements to our fork of `range-v3` to get rid of the owning view at all and have to apply some changes in general to make it possible to mix `range-v3` and `std::ranges` in the same range etc. but that all is nothing you have to worry about.

I used a mix of the solutions: The expression returns an vector<Variable*>, I just appended the target to that vector of pointers, and then passed it through `transform(dereference)`.
I will merge this once the checks have run through, so you don't have to do anything further here.